### ISSUE TITLE
Move Definition Line to Center of Screen after Jump

### ISSUE THREAD

**Issue by invokesus** [ananthakumaran/tide#252]
## Actual Behavior
After invoking `tide-jump-to-definition`, the jump is successful but the definition lies at the bottom of the screen, which forces me to press `C-l` twice to move the line to the top.
Is this by design?

## Expected Behavior
I added `(recenter-top-bottom 1)` to the `tide-jump-to-definition` function which moves the line to the top of the screen. Would you like a PR with this change?

Thanks for this wonderful package.

**Comment by ananthakumaran** [comment]
Please open a PR. `xref` seems to call `(recenter)` after jump, which positions the cursor in the center of the window. Not sure if we should position it in the top or center. I would prefer center as it gives more context. Add the call in `tide-jump-to-filespan` function.

```elisp
(defcustom xref-after-jump-hook '(recenter
                                  xref-pulse-momentarily)
  "Functions called after jumping to an xref."
  :type 'hook)
```

**Comment by josteink** [comment]
> Is this by design?

I'm going to assume this is just incidentally part of the implementation and not really anything anyone has thought through. That is, no existing code was written to specifically get this behaviour.

>  Not sure if we should position it in the top or center. I would prefer center 

Seconding this.

But I'd only like to have this behaviour if we actually need to scroll away from our current point to bring the new definition into view.

**Comment by josteink** [comment]
If there’s no trivial way to do this (I can’t check right now), I guess it’s better to have an explicitly defined, consistent behavior than a implicitly defined inconsistent behavior. 

Consider my comment a wish more than an absolute demand :)

**Comment by invokesus** [comment]
I think you raise a valid point. I tried it out and the recentering was distracting when the jump point was near the initial position. I will look into how to check if the jump point is on screen initially. Right now, I have a sort of workaround where I have disabled recentering if the new position is in the same buffer. 

If there's no easy way to do it, I agree that following `xref`'s implementation might not be a bad idea.  And maybe we can set a variable for disabling it, in case anyone finds it distracting.
